ls8/cpu.py

```diff
@@ -62,4 +80,31 @@ def trace(self):
 
     def run(self):
         """Run the CPU."""
-        pass
+        # It needs to read the memory address that's stored in register `PC`, and store
+        # that result in `IR`, the _Instruction Register_. This can just be a local
+        # variable in `run()`.
+        IR = self.ram[self.pc]
```

Comment:
move this inside of your running loop and then you can avoid typing "self.ram[self.pc]" so many times DRY!

Reply:
oh yes thanks!

Reply:
I was getting an error when I was using the IR variable in the loop. but now it works. thanks